### PR TITLE
PE-19 Extract single merchant details

### PR DESCRIPTION
**Pull request by ltronky**
#### List of Changes
- Add function to retrieve the merchant details by its ID

**Note: Ensure the `discount.staticCode` visibility only to IO app requests will be subject of another PR**

#### Motivation and Context
New feature for CGN operator search API

#### How Has This Been Tested?
Tested locally in dev environment, and with attached unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: GetMerchant/__tests__/handler.test.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+const queryMock = jest.fn().mockImplementation(_ => {});
+
+const cgnOperatorDbMock = { query: queryMock };
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe("GetMerchantHandler", () => {
```
Comment (AleDore) [inline GetMerchant/__tests__/handler.test.ts:15]: In order to improve readability we can move constants definition outside the `describe` section :)

File: GetMerchant/__tests__/handler.test.ts [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+            ],
+            startDate: new Date("2020-01-01"),
+            staticCode: "xxx"
+          }
+        ]
+      });
+    }
+  });
+
+  it("should return a merchant given its ID, also if there is no address associated", async () => {
```
Comment (AleDore) [inline GetMerchant/__tests__/handler.test.ts:137]: ditto :)

**Review by AleDore — COMMENTED** [review]
This is the first part of review

**Comment by ltronky** [comment]
I should have addressed all the suggestions

File: GetMerchant/handler.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  merchantId: string
+) => Promise<ResponseTypes>;
+
+const addressesTask: (
+  cgnOperatorDb: Sequelize,
+  profileId: number
+) => TaskEither<IResponseErrorInternal, ReadonlyArray<AddressModel>> = (
+  cgnOperatorDb: Sequelize,
+  profileId: number
+) =>
```
Comment (AleDore) [inline GetMerchant/handler.ts:57]:
```suggestion
const addressesTask = (
  cgnOperatorDb: Sequelize,
  profileId: number
): TaskEither<IResponseErrorInternal, ReadonlyArray<AddressModel>> =>
```

File: GetMerchant/handler.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    toError
+  ).mapLeft<IResponseErrorInternal>(e => ResponseErrorInternal(e.message));
+
+const discountsTask: (
+  cgnOperatorDb: Sequelize,
+  merchantId: string
+) => TaskEither<IResponseErrorInternal, ReadonlyArray<DiscountModel>> = (
+  cgnOperatorDb: Sequelize,
+  merchantId: string
+) =>
```
Comment (AleDore) [inline GetMerchant/handler.ts:75]:
```suggestion
const discountsTask = (
  cgnOperatorDb: Sequelize,
  merchantId: string
): TaskEither<IResponseErrorInternal, ReadonlyArray<DiscountModel>> =>
```

**Review by AleDore — CHANGES_REQUESTED** [review]

**Review by AleDore — APPROVED** [review]

**Pull request merged**